cs/ObjectPool.cs

```diff
@@ -157,6 +160,8 @@ public void UpdateFactory(ObjectFactory<TSource, TObject> factory)
 
         internal void ReturnObject(PooledObject<TSource, TObject> pooledObject)
         {
+            Contract.Ensures(pooledObject != null);
```

Comment:
depends on compile-time defines. since I don't specify anything it doesn't throw. That's the reason I'm sometimes using throw ArgumentNullException and Contract.EndBlock()